internal/artifact/s3.go

```diff
@@ -24,168 +24,144 @@ const (
 )
 
 type buildkiteEnvProvider struct {
-	retrieved bool
+	next aws.CredentialsProvider
 }
 
-func (e *buildkiteEnvProvider) Retrieve() (credentials.Value, error) {
-	creds := credentials.Value{}
-
-	e.retrieved = false
-
-	creds.AccessKeyID = os.Getenv("BUILDKITE_S3_ACCESS_KEY_ID")
-	if creds.AccessKeyID == "" {
-		creds.AccessKeyID = os.Getenv("BUILDKITE_S3_ACCESS_KEY")
+func (p buildkiteEnvProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
```

Comment:
can we add a test for this function as it does a bunch of comparisons?

Reply:
Added.